### PR TITLE
Remove deprecation warnings on 0.4

### PR DESCRIPTION
**Pull request by garborg**
As a bonus, takes care of the Task switching bug that's keeping DataFrames tests from running on 0.4

### CODE REVIEW

File: test/runtests.jl [hunk 1]
```diff
@@ -66,10 +66,15 @@ close(gzfile)
...
+write(raw_file, zeros(UInt8, 10))
 close(raw_file)
 
-@test_throws ArgumentError gzopen(readall, test_compressed)
+try
+    gzopen(readall, test_compressed)
+    throw(Error("Expecting an ArgumentError or similar"))
+catch e
+    @test isa(e, ArgumentError) || contains(e.msg, "too many arguments")
+end
```
Comment (garborg) [inline test/runtests.jl]: This isn't working on 0.3 or 0.4 on Travis. Locally, it's an `ArgumentError` on 0.4 and and an `Error` with the above message on 0.3. Any hints as to the inconsistency?

Comment (jakebolewski) [inline test/runtests.jl]: Many of the generic error's thrown in base we made more specific as to the actual type of the error (ArgumentError in this case).

Comment (garborg) [inline test/runtests.jl]: I've got that -- just confused why I'm getting those locally and `ZError`s on Travis for both 0.3. and 0.4. Could add more logic in the catch block, just hoping I could avoid haphazardly papering over an issue.

**Comment by garborg** [comment]
Argh, forgot the requisite Compat patch was still waiting on Travis. Restarted -- apologies for the noise.

**Comment by garborg** [comment]
Error on 0.4 belongs to JuliaParser:
```
ERROR: LoadError: LoadError: LoadError: error in method definition: function Base.parse must be explicitly imported to be extended
...
while loading /home/travis/.julia/v0.4/JuliaParser/src/parser.jl, in expression starting on line 1864
```
Passes locally on 0.4.

**Comment by jakebolewski** [comment]
Hmm, fixing this is going to be a bit of a pain.  I don't think I will get to it tonight.

**Comment by garborg** [comment]
Since the tests passed (no `JuliaParser` in the dependency tree) and the hang up was in `Coveralls` submission, does it makes sense to merge this in the meantime so I can tag a patch?

**Comment by jakebolewski** [comment]
Yes, go ahead.

**Comment by garborg** [comment]
I don't have the permissions (to merge here, fine for metadata).

**Comment by garborg** [comment]
Thanks!

**Pull request merged**